salt/metalk8s/orchestrate/deploy_new_node.sls

```diff
@@ -63,7 +63,7 @@ Kill kube-controller-manager on all master nodes:
 
 Register the node into etcd cluster:
   module.run:
-    - metalk8s.add_etcd_node:
+    - metalk8s_etcd.add_etcd_node:
```

Comment:
In my opinion we should have an idempotent state, to not trying to add when node already added
(Not tested with this new version with `python etcd3` but using `etcdctl` if we try to add a node already in the cluster we got an error
```
                                  An exception occurred in this state: Traceback (most recent call last):
                                    File "/usr/lib/python2.7/site-packages/salt/state.py", line 1919, in call
                                      **cdata['kwargs'])
                                    File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1918, in wrapper
                                      return f(*args, **kwargs)
                                    File "/usr/lib/python2.7/site-packages/salt/utils/decorators/__init__.py", line 558, in _decorate
                                      return self._call_function(kwargs)
                                    File "/usr/lib/python2.7/site-packages/salt/utils/decorators/__init__.py", line 263, in _call_function
                                      raise error
                                  CommandExecutionError: etcdctl: membership: peerURL exists
```

If the python module well handle with error for me it's ok to keep a `module.run` if it's not we need a custom state to check list of member and do nothing if member already there

Reply:
>  a custom state to check list of member and do nothing if member already there

This can be racy no?
If possible it's better to do it and handle the error gracefully.

Reply:
Oups, sorry @slaperche-scality I did not see your comment, so I went for the custom state (could be reverted), but to be honest I don't understand how it could be racy as the IP address is uniq per node, I may miss something

Reply:
The node could be added/removed between the time you check and the time you act based on this information.

E.g.:
- your code lists the the cluster members, the node isn't in it
- something/someone add the node to the cluster (someone else also run the state).
- your code adds the node: error!

But maybe such case is not possible with Salt (can't run two states concurrently), then it's fine :slightly_smiling_face: